teachertool/src/components/MakecodeFrame.tsx

```diff
@@ -1,21 +1,28 @@
 /// <reference path="../../../built/pxteditor.d.ts" />
-import * as React from "react";
+import { useContext } from "react";
 import { setEditorRef } from "../services/makecodeEditorService";
-interface MakeCodeFrameProps {
-    pageSourceUrl: string;
-    tutorialEventHandler?: (event: pxt.editor.EditorMessageTutorialEventRequest) => void;
-}
+import { AppStateContext } from "../state/appStateContext";
+import { createIFrameUrl } from "../utils";
 
-export const MakeCodeFrame: React.FC<MakeCodeFrameProps> =
-    ( { pageSourceUrl} ) => {
+interface MakeCodeFrameProps {}
+export const MakeCodeFrame: React.FC<MakeCodeFrameProps> = () => {
+    const { state: teacherTool } = useContext(AppStateContext)
 
     const handleIFrameRef = (el: HTMLIFrameElement | null) => {
         setEditorRef(el ?? undefined);
     }
 
     /* eslint-disable @microsoft/sdl/react-iframe-missing-sandbox */
-    return <div className="makecode-frame-outer" style={{ display: "block" }}>
-        <iframe className="makecode-frame" src={pageSourceUrl} title={"title"} ref={handleIFrameRef}></iframe>
-    </div>
+    return (
+        <div className="makecode-frame-outer" style={{ display: "block" }}>
+            {teacherTool.projectMetadata && (
```

Comment:
Similar to my comment above, I think you can just have this check in `App.tsx`.

Reply:
My hesitation with this one is that if we ever end up using this component in more than one place, we'd have to put the check in every time we specify it. Whereas inside the component, we have full context, knowing we're dependent specifically on this field in the state and that it doesn't make sense to render the component without it (in fact, the app would look like it was broken).

A sample scenario where this sort of thing might be relevant, imagine we decide to throw in a spinner or some kind of loading/placeholder message when projectMetadata isn't set. If we keep this check inside the component, we only have to update it here. However, if we push it out, then we have to update all the places that have used the component.

In this context, it's a little academic since I don't think we'll be using MakecodeFrame in more than one place in the app, but it's sticking to a convention that I think makes sense on the whole...